winch/codegen/src/isa/x64/masm.rs

```diff
@@ -98,6 +88,70 @@ impl Masm for MacroAssembler {
         self.asm.trapif(IntCmpKind::GtU, TrapCode::StackOverflow);
     }
 
+    fn save_clobbers(&mut self, clobbers: &[(Reg, OperandSize)]) {
+        // Determine how much space we need for the clobbers
+        let clobbered_size = align_to(
+            clobbers
+                .iter()
+                .fold(0u32, |total, (reg, _)| match reg.class() {
+                    RegClass::Int => total + 8,
+                    RegClass::Float => align_to(total, 16) + 16,
+                    RegClass::Vector => unimplemented!(),
+                }),
+            16,
+        );
+
+        // Emit unwind info.
+        if self.shared_flags.unwind_info() {
+            self.asm.emit_unwind_inst(UnwindInst::DefineNewFrame {
+                offset_upward_to_caller_sp: Self::ABI::arg_base_offset().try_into().unwrap(),
+                offset_downward_to_clobbers: clobbered_size,
+            })
+        }
+
+        self.reserve_stack(clobbered_size);
+
+        let mut off = 0;
+        for &(reg, size) in clobbers {
+            // Align the current offset
+            off = align_to(off, size.bytes());
+
+            // Emit the store
+            let addr = self.address_at_sp(SPOffset::from_u32(off));
+            self.store(RegImm::Reg(reg), addr, size);
+
+            // Emit unwinding info, if necessary
+            if self.shared_flags.unwind_info() {
+                self.asm.emit_unwind_inst(UnwindInst::SaveReg {
+                    clobber_offset: off,
+                    reg: reg.into(),
+                });
+            }
+
+            // Increment the offset
+            off += size.bytes();
+        }
+
+        // debug_assert_eq!(align_to(off, 16), clobbered_size);
```

Comment:
Do we still want to keep this `debug_assert_eq`?

Reply:
I think so, after the change above. I'll uncomment it :)